services/horizon/internal/codes/main.go

```diff
@@ -491,6 +491,12 @@ func ForOperationResult(opr xdr.OperationResult) (string, error) {
 		ic = ir.MustEndSponsoringFutureReservesResult().Code
 	case xdr.OperationTypeRevokeSponsorship:
 		ic = ir.MustRevokeSponsorshipResult().Code
+	case xdr.OperationTypeClawback:
+		ic = ir.MustClawbackResult().Code
+	case xdr.OperationTypeClawbackClaimableBalance:
+		ic = ir.MustClawbackClaimableBalanceResult().Code
+	case xdr.OperationTypeSetTrustLineFlags:
+		ic = ir.MustSetTrustLineFlagsResult().Code
```

Comment:
This could had gone easily unnoticed (I just happened to test against the opcode).

We have a few functions in which we return a default value if the type value isn't found. Instead I think we should have something like `MaxKnownOperationType`, which we change when new operations are added (`MaxKnownOperationType` would had been updated to  `xdr.OperationTypeSetTrustLineFlags` when changing the for CAP 35). In that way we would had been able to do:

```suggestion
	  case ir.Type <= MaxKnownOperationType:
	    return "", errors.New("internal error: not implemented!")
```

Then again, we could also forget to bump `MaxKnownOperationType` when adding new operations 🤷 

Thoughts?

Reply:
Why not `default:` case? We use it in other places, we should here too.

Reply:
I thought there was no default error in order to be forward compatible.

Reply:
But, to be fair, we already error in other places, which will stop ingestion (notably, the stats update processing).